hal/analogin_api.h

```diff
@@ -35,9 +35,29 @@ typedef struct analogin_s analogin_t;
 
 /**
  * \defgroup hal_analogin Analogin hal functions
+ *
+ * # Defined behaviour
+ * * The function ::analogin_init initializes the analogin peripheral
+ * * The function ::analogin_read reads the input voltage, represented as a float in the range [0.0, 1.0]
+ * * The function ::analogin_read_u16 reads the value from analogin pin, represented as an unsigned 16bit value
```

Comment:
A couple of good things to add:
- Required tolerance. Maybe require the ADC to be +-20% accurate or something
- Voltage range - maybe 0.0 is ground and 1.0 is Vcc?